File: circlesum.py
def nowrapsum(numbers):
    n = len(numbers)
    max = 0
    curr = 0
    for i in range(1, n):
        curr += numbers[i]
        if(curr < 0):
            curr = 0
        if(curr > max):
            max = curr
    return max

def main():
    try:
        filein = open("circlesum.in", "r")
        file = open("circlesum.out", "w")
        print("files opened")
        numbers = filein.readlines()
        numbers = [int(i) for i in numbers]
        #numbers = [28, 5, 10, 15, -2, -6, 45, -2, 10, 5, 2, 1, 20, 15, 14, 30, -2, 1, 35, 63, 2, 4 ,-7, 1, 6, 8, 12, 45, -6]
        n = len(numbers)
        nowrapmax = nowrapsum(numbers)
        print("function called")
        max = 0
        for i in range(1, n):
            max += numbers[i]
            numbers[i] = -numbers[i]
        max = max + nowrapsum(numbers)
        if max > nowrapmax:
            file.write(str(max))
        else:
            file.write(str(nowrapmax))
        file.close()
    except:
        print("exception")
if __name__=="__main__":
    main()

File: circlesum.java
import java.util.*;
import java.io.*;
public class circlesum {
public static void main(String[] args){
    try{
	FileReader fr = new FileReader("circlesum.in");
	BufferedReader br = new BufferedReader(fr);
        FileWriter out = new FileWriter("circlesum.out");
        System.out.println("files opened");
	int seeds = Integer.parseInt(br.readLine());
	System.out.println("first number parsed");
	int[] numbers = new int[seeds];
	System.out.println("array made");
	numbers[0] = seeds;
	for(int x=1; x < seeds + 1; x++) 
		numbers[x] = Integer.parseInt(br.readLine());
	System.out.println("array finished");
        System.out.println("files converted");
        int begcirclength = 0;
        int index = 1;
        // finds sum of beginning positive numbers, added on to end of array sum
        System.out.println(numbers);
        while(index < numbers.length){
            if(numbers[index] < 0)
                break;
            begcirclength += numbers[index];
            index ++;
	}
        System.out.println("begcirclength: " + begcirclength);
        int max = begcirclength;
        int cursum = 0;
        for (int i=index; i < seeds; i++) {
            if(numbers[i] > 0)
                cursum += numbers[i];
            else{
                if(cursum > max)
                    max = cursum;
                cursum = 0;
	    }
            System.out.println("cursum: " + cursum+ " after number: "+ numbers[i]);
	}
        if(cursum > 0){
            cursum += begcirclength;
            if(cursum > max)
                max = cursum;
	}
        String s = String.valueOf(max);
        out.write(s);
        out.close();
	}
    catch(Exception e) {
        System.out.println("exception");
    }
}
}
